File: app/src/main/java/com/u1fukui/bbs/ui/main/mypage/HistoryThreadListFragment.java
package com.u1fukui.bbs.ui.main.mypage;

import com.u1fukui.bbs.repository.HistoryThreadListRepository;
import com.u1fukui.bbs.repository.ThreadListRepository;
import com.u1fukui.bbs.ui.main.BaseThreadListFragment;

public class HistoryThreadListFragment extends BaseThreadListFragment {

    public static final HistoryThreadListFragment newInstance() {
        return new HistoryThreadListFragment();
    }

    public HistoryThreadListFragment() {
    }

    @Override
    protected ThreadListRepository getRepository() {
        return new HistoryThreadListRepository();
    }
}

File: app/src/main/java/com/u1fukui/bbs/ui/main/mypage/HistoryThreadListFragment.kt
package com.u1fukui.bbs.ui.main.mypage

import com.u1fukui.bbs.api.ThreadListApi
import com.u1fukui.bbs.repository.HistoryThreadListRepository
import com.u1fukui.bbs.ui.main.BaseThreadListFragment
import javax.inject.Inject

class HistoryThreadListFragment : BaseThreadListFragment() {

    @Inject
    lateinit var threadListApi: ThreadListApi

    override fun getRepository() = HistoryThreadListRepository(threadListApi)

    companion object {

        fun newInstance() = HistoryThreadListFragment()
    }
}
